### PR TITLE
Fix sphinx-build warnings

### LINKED ISSUE #1510: Fix sphinx-build warnings
Currently, generating documentation locally with Sphinx involves activating the ``nest-doc`` environment and entering ``make html``: https://nest-simulator.readthedocs.io/en/latest/documentation_workflow/user_documentation_workflow.html

The HTML pages are rendered well. However, numerous warnings are emitted in the process. These relate to formatting issues, broken links, incomplete references, etc.

We have begun working on these warnings in @steffengraber's ``cleanup-doc`` branch.

### PR DESCRIPTION
**Pull request by sarakonradi**
Fixes #1510

Currently, numerous warnings are emitted when generating HTML with Sphinx locally. 

The majority of these warnings are related to the model files and could disappear with the merge of #1374. This is because files (e.g., ``doc/models/create_models.rst``) that result in toctree warnings will anyhow be deleted in the other PR.

Given this information, the current PR has begun to tackle the other documentation-related warnings.

### CODE REVIEW

**Comment by terhorstd** [comment]
this would be nice to have for nest-3, but is in no way mandatory. removing from milestone.

**Comment by jougs** [comment]
During the current hackathon we decided that we should merge this as soon as possible to avoid endless merge conflicts and duplication of work in other branches. New warnings can then be addressed in new pull requests.

File: doc/contribute/mailing_list_guidelines.rst [hunk 1]
```diff
@@ -1,3 +1,5 @@
+:orphan:
+
 Guidelines for contributing to the mailing list
 =================================================
```
Comment (jougs) [inline doc/contribute/mailing_list_guidelines.rst:4]:
```suggestion
===============================================
```

File: doc/contribute/templates_styleguides/example_template.rst [hunk 2]
```diff
@@ -1,3 +1,5 @@
+:orphan:
+
 PyNEST example template
 ==========================
```
Comment (jougs) [inline doc/contribute/templates_styleguides/example_template.rst:4]:
```suggestion
=======================
```

File: doc/contribute/templates_styleguides/pyapi_template.rst [hunk 3]
```diff
@@ -1,3 +1,5 @@
+:orphan:
+
 PyNEST API template
 =====================
```
Comment (jougs) [inline doc/contribute/templates_styleguides/pyapi_template.rst:4]:
```suggestion
===================
```

File: doc/glossary.rst [hunk 4]
```diff
@@ -57,6 +57,8 @@ Common abbreviations in NEST
    virtual process
 
 Physical units and variable names used for NEST parameters
+:orphan:
+
```
Comment (jougs) [inline doc/glossary.rst:61]: This should be moved to the beginning of the file. The way it is now breaks the formatting of the headline

File: doc/glossary.rst [hunk 5]
```diff
@@ -57,6 +57,8 @@ Common abbreviations in NEST
...
 -------------------------------------------------------------
```
Comment (jougs) [inline doc/glossary.rst:62]:
```suggestion
----------------------------------------------------------
```

**Review by jougs — CHANGES_REQUESTED** [review]
Thanks for addressing these. I think we can even do without the second review by @stinebuu, so I'm removing the request for review.

**Comment by jougs** [comment]
Also, please resolve the conflicts. Thanks!

**Review by jougs — APPROVED** [review]
Happy if Travis is. Thanks for taking care!

**Pull request merged**